src/scripts/main.js

```diff
@@ -1,3 +1,267 @@
+    messageForWinner.classList.add('hidden');
+  }
+
+  score = 0;
+  pointsCounter.textContent = score.toString();
+  generateNumber();
+});
+
+function changeButtonClass() {
+  theOnlyButton.textContent = 'Restart';
+  theOnlyButton.classList.remove('start');
+  theOnlyButton.classList.add('restart');
+}
+
+function hideStartMessage() {
+  document.querySelector('.message-start').classList.add('hidden');
+}
+
+function generateNumber() {
+  const isEmptyCell = [...allCells].some(cell => cell.textContent === '');
+
+  if (!isEmptyCell) {
+    messageForLooser.classList.remove('hidden');
+
+    return;
+  }
+
+  let found = false;
+
+  while (!found) {
+    const horizontallyCell = Math.floor(Math.random() * boardSize);
+    const verticallyCell = Math.floor(Math.random() * boardSize);
+    const cellForNewNumber = board.rows[horizontallyCell].cells[verticallyCell];
+
+    if (cellForNewNumber.textContent === '') {
+      const newNumber = Math.random() >= 0.9 ? 4 : 2;
+
+      cellForNewNumber.textContent = `${newNumber}`;
+      updateClassList(cellForNewNumber);
+      found = true;
+    }
+  }
+}
+
+document.addEventListener('keyup', (e) => {
+  hideStartMessage();
+
+  if (e.code === 'Numpad8') {
+    moveNumbersUp();
+    generateNumber();
+  }
+
+  if (e.code === 'Numpad6') {
+    moveNumbersRight();
+    generateNumber();
+  }
+
+  if (e.code === 'Numpad4') {
+    moveNumbersLeft();
+    generateNumber();
+  }
+
+  if (e.code === 'Numpad2') {
+    moveNumbersDown();
+    generateNumber();
+  }
+
+  pointsCounter.textContent = `${score}`;
+  changeButtonClass();
+});
+
+function makeArrayOfNumbersInColumns() {
+  const array = [];
+  let step = 0;
+
+  do {
+    const ar = [];
+
+    for (let i = step; i < allCells.length; i += boardSize) {
+      if (allCells[i].textContent !== '') {
+        ar.push(Number(allCells[i].textContent));
+      }
+    }
+
+    array.push(ar);
+    step++;
+  } while (step !== 4);
+
+  return array;
+}
+
+function makePairsOfNumbers(array) {
+  for (let x = 0; x < array.length; x++) {
+    for (let i = 0; i < array[x].length; i++) {
+      if (array[x][i] === array[x][i + 1]) {
+        array[x][i] *= 2;
+
+        if (array[x][i] === winnerScore) {
+          messageForWinner.classList.remove('hidden');
+        }
+
+        array[x][i + 1] = 0;
+        score += array[x][i];
+      }
+
+      array[x] = array[x].filter(a => a !== 0);
+    }
+  }
+
+  return array;
+}
+
+function matchNumbersOpp(array) {
+  for (let x = 0; x < array.length; x++) {
+    for (let i = array[x].length - 1; i >= 0; i--) {
+      if (array[x][i] === array[x][i - 1]) {
+        array[x][i] *= 2;
+
+        if (array[x][i] === winnerScore) {
+          messageForWinner.classList.remove('hidden');
+        }
+
+        array[x][i - 1] = 0;
+        score += array[x][i];
+      }
+    }
+
+    array[x] = array[x].filter(num => num !== 0);
+  }
+
+  return array;
+}
+
+function makeArrayOfNumbersInRows() {
+  const ar = [];
```

Comment:
```suggestion
  const arr = [];
```
Don't name arrays `ar`. It's incorrect.